src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -521,43 +803,36 @@ function AnalyticsService:LogPlayerEvent(data: PlayerEvent): { [any]: any }
 		elseif data.value ~= nil and typeof(data.value) ~= "number" then
 			return reject("AnalyticsService.LogPlayerEvent - value must be a number")
 		end
-		
-		-- Trim trailing colon
-		if string.sub(data.event, #data.event) == ":" then
-			data.event = string.sub(data.event, 1, #data.event - 1)
-		end
-		
-		GameAnalytics:addDesignEvent(data.userId, {
+
+		-- Sanitize event string to prevent errors
+		data.event = string.gsub(data.event, "[^%w%-_%.%s:]", "_")
+
+		-- Trim trailing colon and spaces
+		data.event = string.gsub(data.event, ":%s*$", "")
+
+		self:_wrapper("addDesignEvent", data.userId, {
 			eventId = data.event,
-			value = data.value
+			value = data.value,
 		})
-		
-		return resolve()
+
+		resolve()
```

Comment:
```suggestion
		return resolve()
```

The linter/type checker complains if there's a return type defined for a function, but the promise is resolved/rejected without returning the function (For some reason)? It's an invalid error since we know we're returning a promise, but the code checker flags it for some reason.